exercises/practice/rational-numbers/src/RationalNumbers.hs

```diff
@@ -0,0 +1,46 @@
+module RationalNumbers
+(Rational,
+ abs,
+ reduce,
+ add,
+ sub,
+ mul,
+ div,
+ exprational,
+ expreal,
```

Comment:
Name suggestions:

- `expRational`, or plain `exp`
- `expReal`

Reply:
I've taken these identifiers from `canonical-data.schema.json` from problem description. I'm unsure to which extend am I allowed to deviate from that file. If I could choose identifiers, I'd use `pow` and `exp` ;-)

I have also a different question in this context and I'm not sure how to proceed:

The instructions require the support for three types of exponentiation:

a) Exponentiation of a rational number to an integer power with two sub-cases (positive and negative exponent)
b) Exponentiation of a rational number to a real number
c) Exponentiation of a real number to a rational number

There are however no tests for the case b) in the problem specification. If I want to cover this case as well, shall I add new tests also there, or only in the Haskell track?

Reply:
> I'm unsure to which extend am I allowed to deviate from that file.

There is not set limit. Names are routinely tailored to language tracks.

> The instructions require the support for three types of exponentiation: [&hellip;]

I think we are allowed to deviate from the canonical instructions, just like we are allowed to deviate from the canonical tests. @ErikSchierboom is this correct / what do you think?

Reply:
That is correct